crates/bevy_pbr/src/render/prepass.rs

```diff
@@ -0,0 +1,553 @@
+    }
+}
+
+#[derive(Resource)]
+pub struct PrepassPipeline<M: Material> {
+    pub view_layout: BindGroupLayout,
+    pub mesh_layout: BindGroupLayout,
+    pub skinned_mesh_layout: BindGroupLayout,
+    pub material_layout: BindGroupLayout,
+    pub material_vertex_shader: Option<Handle<Shader>>,
+    pub material_fragment_shader: Option<Handle<Shader>>,
+    pub material_pipeline: MaterialPipeline<M>,
+    _marker: PhantomData<M>,
+}
+
+impl<M: Material> FromWorld for PrepassPipeline<M> {
+    fn from_world(world: &mut World) -> Self {
+        let render_device = world.resource::<RenderDevice>();
+        let asset_server = world.resource::<AssetServer>();
+
+        let view_layout = render_device.create_bind_group_layout(&BindGroupLayoutDescriptor {
+            entries: &[
+                // View
+                BindGroupLayoutEntry {
+                    binding: 0,
+                    visibility: ShaderStages::VERTEX | ShaderStages::FRAGMENT,
+                    ty: BindingType::Buffer {
+                        ty: BufferBindingType::Uniform,
+                        has_dynamic_offset: true,
+                        min_binding_size: Some(ViewUniform::min_size()),
+                    },
+                    count: None,
+                },
+            ],
+            label: Some("prepass_view_layout"),
+        });
+
+        let mesh_pipeline = world.resource::<MeshPipeline>();
+
+        PrepassPipeline {
+            view_layout,
+            mesh_layout: mesh_pipeline.mesh_layout.clone(),
+            skinned_mesh_layout: mesh_pipeline.skinned_mesh_layout.clone(),
+            material_vertex_shader: match M::prepass_vertex_shader() {
+                ShaderRef::Default => None,
+                ShaderRef::Handle(handle) => Some(handle),
+                ShaderRef::Path(path) => Some(asset_server.load(path)),
+            },
+            material_fragment_shader: match M::prepass_fragment_shader() {
+                ShaderRef::Default => None,
+                ShaderRef::Handle(handle) => Some(handle),
+                ShaderRef::Path(path) => Some(asset_server.load(path)),
+            },
+            material_layout: M::bind_group_layout(render_device),
+            material_pipeline: world.resource::<MaterialPipeline<M>>().clone(),
+            _marker: PhantomData,
+        }
+    }
+}
+
+impl<M: Material> SpecializedMeshPipeline for PrepassPipeline<M>
+where
+    M::Data: PartialEq + Eq + Hash + Clone,
+{
+    type Key = MaterialPipelineKey<M>;
+
+    fn specialize(
+        &self,
+        key: Self::Key,
+        layout: &MeshVertexBufferLayout,
+    ) -> Result<RenderPipelineDescriptor, SpecializedMeshPipelineError> {
+        let mut bind_group_layout = vec![self.view_layout.clone()];
+        let mut shader_defs = Vec::new();
+        let mut vertex_attributes = Vec::new();
+
+        // NOTE: Eventually, it would be nice to only add this when the shaders area overloaded.
```

Comment:
Essentially, when using the default prepass, it doesn't need a material, but because the bind group index are hardcoded in the shader I need to pass something at the bind group index 1. I'm not entirely sure of the actual cost of doing that, so it's possibly a non-issue, but it essentially means that there's an unnecessary bind group when using the default prepass.

Reply:
```suggestion
        // NOTE: Eventually, it would be nice to only add this when the shaders are overloaded by the Material.
```